code/datums/abilities/revenant.dm

```diff
@@ -172,17 +172,17 @@
 		src.wraith = null
 		return
 
-	OnLife()
+	OnLife(var/mult)
 		if (!src.wraith)
 			return
 		if (ghoulTouchActive)
-			ghoulTouchActive--
+			ghoulTouchActive -= mult
 			if (!ghoulTouchActive)
 				owner.show_message("<span class='alert'>You are no longer empowered by the netherworld.</span>")
```

Comment:
Need to bound ghoulTouchActive to [0-inf), or update `melee_attack_proc.dm` and here to check >0